scylla/src/transport/locator/mod.rs

```diff
@@ -549,3 +553,240 @@ impl<'a> Iterator for ReplicaSetIterator<'a> {
+            } => {
+                // Clippy can't check that in Eq and Hash impls we don't actually use any field with interior mutability
+                // (in Node only `down_marker` is such, being an AtomicBool).
+                // https://rust-lang.github.io/rust-clippy/master/index.html#mutable_key_type
+                #[allow(clippy::mutable_key_type)]
+                let mut all_replicas: HashSet<&'a Arc<Node>> = HashSet::new();
+                for (datacenter, repfactor) in datacenter_repfactors.iter() {
+                    all_replicas.extend(
+                        locator
+                            .get_network_strategy_replicas(token, datacenter, *repfactor)
+                            .iter(),
+                    );
+                }
+                // It's no use returning a node that was already picked.
+                all_replicas.remove(picked);
+
+                let mut replicas_ordered = vec![];
+                let nodes_on_ring = locator.replication_data.get_global_ring().ring_range(token);
+                for node in nodes_on_ring {
+                    if all_replicas.is_empty() {
+                        // All replicas were put in order.
+                        break;
+                    }
+                    if all_replicas.remove(node) {
+                        replicas_ordered.push(node);
+                    }
+                }
+                assert!(
+                    all_replicas.is_empty(),
+                    "all_replicas somehow contained a node that wasn't present in the global ring!"
+                );
+
+                *self = Self::ComputedFallback {
+                    replicas: ReplicasArray::Owned(replicas_ordered),
+                    idx: 0,
+                };
+                self.next()
+            }
+            Self::ComputedFallback {
+                ref replicas,
+                ref mut idx,
+            } => {
+                if let Some(replica) = replicas.get(*idx) {
+                    *idx += 1;
+                    Some(replica)
+                } else {
+                    None
+                }
+            }
+        }
+    }
+}
+
+impl<'a> Iterator for ReplicasOrderedIterator<'a> {
+    type Item = NodeRef<'a>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        match self {
+            ReplicasOrderedIterator::SimpleStrategy { replica_set_iter } => replica_set_iter.next(),
+            ReplicasOrderedIterator::NetworkTopologyStrategy {
+                replicas_ordered_iter,
+            } => replicas_ordered_iter.next(),
+        }
+    }
+}
+
+impl<'a> IntoIterator for ReplicasOrdered<'a> {
+    type Item = NodeRef<'a>;
+    type IntoIter = ReplicasOrderedIterator<'a>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        let Self { replica_set } = self;
+        match replica_set.inner {
+            ReplicaSetInner::Plain(_) | ReplicaSetInner::FilteredSimple { .. } => {
+                Self::IntoIter::SimpleStrategy {
```

Comment:
`SimpleStrategy` is a bad name for this variant - `ReplicaSetInner::Plain` may carry NTS replicas restricted to a single DC.

Reply:
I wasn't aware of that.

Reply:
I've changed it as follows:
`SimpleStrategy` -> `AlreadyRingOrdered`
`NetworkTopologyStrategy` -> PolyDatacenterNTS`
is it OK now?

Reply:
It's ok, quite creative use of a `Poly` prefix:))

Reply:
Ancient Greek for the rescue!